interop-testing/src/main/java/io/grpc/testing/integration/XdsTestClient.java

```diff
@@ -0,0 +1,339 @@
+                try {
+                  client.stop();
+                } catch (Exception e) {
+                  e.printStackTrace();
+                }
+              }
+            });
+    client.run();
+  }
+
+  private void parseArgs(String[] args) {
+    boolean usage = false;
+    for (String arg : args) {
+      if (!arg.startsWith("--")) {
+        System.err.println("All arguments must start with '--': " + arg);
+        usage = true;
+        break;
+      }
+      String[] parts = arg.substring(2).split("=", 2);
+      String key = parts[0];
+      if ("help".equals(key)) {
+        usage = true;
+        break;
+      }
+      if (parts.length != 2) {
+        System.err.println("All arguments must be of the form --arg=value");
+        usage = true;
+        break;
+      }
+      String value = parts[1];
+      if ("num_channels".equals(key)) {
+        numChannels = Integer.valueOf(value);
+      } else if ("print_response".equals(key)) {
+        printResponse = Boolean.valueOf(value);
+      } else if ("qps".equals(key)) {
+        qps = Integer.valueOf(value);
+      } else if ("rpc_timeout_sec".equals(key)) {
+        rpcTimeoutSec = Integer.valueOf(value);
+      } else if ("server".equals(key)) {
+        server = value;
+      } else if ("stats_port".equals(key)) {
+        statsPort = Integer.valueOf(value);
+      } else {
+        System.err.println("Unknown argument: " + key);
+        usage = true;
+        break;
+      }
+    }
+
+    if (usage) {
+      XdsTestClient c = new XdsTestClient();
+      System.err.println(
+          "Usage: [ARGS...]"
+              + "\n"
+              + "\n  --num_channels=INT     Default: "
+              + c.numChannels
+              + "\n  --print_response=BOOL  Write RPC response to stdout. Default: "
+              + c.printResponse
+              + "\n  --qps=INT              Qps per channel. Default: "
+              + c.qps
+              + "\n  --rpc_timeout_sec=INT  Per RPC timeout seconds. Default: "
+              + c.rpcTimeoutSec
+              + "\n  --server=host:port     Address of server. Default: "
+              + c.server
+              + "\n  --stats_port=INT       Port to expose peer distribution stats service. "
+              + "Default: "
+              + c.statsPort);
+      System.exit(1);
+    }
+  }
+
+  private void run() {
+    statsServer = NettyServerBuilder.forPort(statsPort).addService(new XdsStatsImpl()).build();
+    boolean success = false;
+    try {
+      statsServer.start();
+      List<ManagedChannel> chans = new ArrayList<>();
+      for (int i = 0; i < numChannels; i++) {
+        chans.add(NettyChannelBuilder.forTarget(server).usePlaintext().build());
+      }
+      exec = MoreExecutors.listeningDecorator(Executors.newSingleThreadScheduledExecutor());
+      try {
+        runQps(chans, exec);
+        success = true;
+      } finally {
+        for (ManagedChannel chan : chans) {
+          chan.shutdownNow();
```

Comment:
nit: This should also be done in ShutdownHook, otherwise it may not be trigger if you press Ctrl-C.

Reply:
done